src/program.hpp

```diff
@@ -124,6 +128,11 @@ struct constant_data_buffer_info {
     std::vector<char> data;
 };
 
+using kernel_image_channel_metadata_map =
+    std::unordered_map<uint32_t, std::pair<uint32_t, uint32_t>>;
```

Comment:
I think I'd rather we use introduce a struct for the metadata itself straight away. This would make the mechanism more extensible and improve readability (this results in code like  `second.first` and `second.second` in places already).

Reply:
Done